tests/network.rs

```diff
@@ -183,22 +187,13 @@ impl Node {
         self.table.is_close(name)
     }
 
-    // Connect to the node with the given name and endpoint.
-    // This adds the new node to our routing table.
-    fn connect(&mut self, their_name: &XorName, their_endpoint: Endpoint) {
-        let _ = self.table.add(their_name.clone());
-        let _ = self.connections.insert(their_name.clone(), Connection(their_endpoint));
-    }
-
     fn send_message(&mut self, mut message: Message, handle: bool) -> Vec<Action> {
         let mut actions = Vec::new();
 
         message.route.push(self.name.clone());
 
         let count = self.message_stats.get_received(message.id).saturating_sub(1);
```

Comment:
(To clarify: If we're sending this message ourselves, `count` should be `0`. If we're relaying a copy of a message that we haven't seen before, it should also be `0`. If this is the `n`-th copy we're relaying, it should be `n - 1`.)